core/src/main/scala/kafka/log/LogValidator.scala

```diff
@@ -414,7 +414,8 @@ private[log] object LogValidator extends Logging {
 
       try {
         val recordErrors = new ArrayBuffer[ApiRecordError](0)
-        for ((record, batchIndex) <- batch.asScala.view.zipWithIndex) {
+        var batchIndex = 0
+        for (record <- recordsIterator.asScala) {
```

Comment:
why not using ```zipWithIndex```?
```scala
        recordsIterator.asScala.zipWithIndex.foreach {
          case (record, batchIndex) =>
```

The ```zipWithIndex``` on scala iterator is a wrap so it should not cause performance regression too much.